From 85d807fa4c7de7767c0ba001ccc1b6a68f6cfb17 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:24 -0400
Subject: [usb] usbtest: bmAttributes would better be masked

Message-id: <1410887463-20674-13-git-send-email-dzickus@redhat.com>
Patchwork-id: 95464
O-Subject: [RHEL7 PATCH 012/591] usb: usbtest: bmAttributes would better be masked
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9a37a50349c8d6a7adcee21cefae407fb2f8c623
Author: Huang Rui <ray.huang@amd.com>
Date:   Tue Sep 24 00:03:43 2013 +0800

    usb: usbtest: bmAttributes would better be masked

    When transfer type is isochronous, the other bits (bits 5..2) of
    bmAttributes in endpoint descriptor might not be set zero. So it's better
    to use usb_endpoint_type routine to mask bmAttributes with
    USB_ENDPOINT_XFERTYPE_MASK to judge the transfter type later.

    Signed-off-by: Huang Rui <ray.huang@amd.com>
    Acked-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/misc/usbtest.c b/drivers/usb/misc/usbtest.c
index aa28ac8..3e91d3e 100644
--- a/drivers/usb/misc/usbtest.c
+++ b/drivers/usb/misc/usbtest.c
@@ -120,7 +120,7 @@ get_endpoints(struct usbtest_dev *dev, struct usb_interface *intf)
    struct usb_host_endpoint *e;
 
    e = alt->endpoint + ep;
-   switch (e->desc.bmAttributes) {
+   switch (usb_endpoint_type(&e->desc)) {
    case USB_ENDPOINT_XFER_BULK:
     break;
    case USB_ENDPOINT_XFER_ISOC:
-- 
1.7.1